lib/src/backend.rs

```diff
@@ -43,7 +43,7 @@ macro_rules! id_type {
             #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Hash)]
             $vis struct $name(Vec<u8>);
         }
-        impl_id_type!($name);
+        crate::backend::impl_id_type!($name);
```

Comment:
nit: I slightly prefer `$crate` (though it wouldn't matter for crate-local macros)

And let's fully spell `std::fmt::Debug`/`Formatter`/`Error` to not rely on names imported at the call site.

Reply:
Good idea. Done.